pageserver/src/tenant/timeline.rs

```diff
@@ -3024,6 +3030,41 @@ impl Timeline {
             .await?;
 
             keyspace.remove_overlapping_with(&completed);
+
+            // Do not descend into the ancestor timeline for aux files.
+            // We don't return a blanket [`GetVectoredError::MissingKey`] to avoid
+            // stalling compaction.
+            // TODO(chi): this will need to be updated for aux files v2 storage
+            if keyspace.overlaps(&NON_INHERITED_RANGE) {
+                let removed = keyspace.remove_overlapping_with(&KeySpace {
+                    ranges: vec![NON_INHERITED_RANGE],
+                });
+
+                for range in removed.ranges {
+                    let mut key = range.start;
+                    while key < range.end {
+                        reconstruct_state.on_key_error(
```

Comment:
I'm wondering about this part. Right now, `NON_INHERITED_RANGE` is just one key so this loop is trivial. However, with aux file storage v2 that range would become quite large (120 bits I guess?) and also be nonconvex (have gaps), so if someone tried reading the aux file range, this loop over the entire not found range might turn into a footgun.

Reply:
Thanks, that makes sense.